src/V3AstNodeExpr.h

```diff
@@ -1932,7 +1932,7 @@ class AstStructSel final : public AstNodeExpr {
     void name(const string& name) override { m_name = name; }
     string emitVerilog() override { V3ERROR_NA_RETURN(""); }
     string emitC() override { V3ERROR_NA_RETURN(""); }
-    bool cleanOut() const override { return false; }
+    bool cleanOut() const override { return VN_IS(fromp()->dtypep()->skipRefp(), StructDType); }
```

Comment:
I thought that it is always a struct, but when I changed it, I noticed that it may be also a union.
If members of a union have different sizes and the shorter member is selected, the bit mask is needed to clean the bits that are used only by the wider members.

Reply:
Done, thanks for the review.